### PR TITLE
Fix New Instance Forge Version Sorting

### PR DESCRIPTION
**Pull request by Mohron**
See https://ptb.discordapp.com/channels/398091532881756161/692020276644282418/692033268047937576

### CODE REVIEW

**Comment by furtabs** [comment]
Seems Fine to Me

**Comment by Mohron** [comment]
How much do we trust the data though? Because if we don't, the sort method isn't robust.

I didn't feel like it warranted the extra dependency that was used in master though.

File: src/common/utils/index.js [hunk 1]
```diff
@@ -6,6 +6,17 @@ export const sortByDate = (a, b) => {
   return dateB - dateA;
 };
 
+export function sortByVersion(a, b) {
```
Comment (blarfoon) [inline src/common/utils/index.js]: This is pretty specific to forge, I would change the function name

File: src/common/utils/index.js [hunk 2]
```diff
@@ -6,6 +6,17 @@ export const sortByDate = (a, b) => {
...
+  const versionA = a.split('.');
+  const versionB = b.split('.');
+
+  for (let i = 0; i < versionA.length; i += 1) {
+    if (versionA[i] !== versionB[i]) {
+      return versionB[i] - versionA[i];
+    }
+  }
```
Comment (blarfoon) [inline src/common/utils/index.js]: Sorting functions always need a return. What if this condition is never true?

**Review by blarfoon — CHANGES_REQUESTED** [review]

File: src/common/utils/index.js (hunk 1 above)
Comment (Mohron) [reply src/common/utils/index.js]: It should work for any `.` separated version string but I renamed it nonetheless

File: src/common/utils/index.js [hunk 3]
```diff
@@ -6,6 +6,24 @@
...
+      const verNumA = Number(versionA[i]) || 0;
+      const verNumB = Number(versionB[i]) || 0;
+
+      if (verNumA !== verNumB) {
+        return verNumB - verNumA;
+      }
+    }
+  }
+
+  return 0
```
Comment (killpowa) [inline src/common/utils/index.js]: Codacy Issue found: [Insert `;`](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5243672)

**Comment by killpowa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5243672&bid=17187151)

File: src/common/utils/index.js [hunk 4]
```diff
@@ -6,6 +6,24 @@
   return dateB - dateA;
 };
 
+export function sortByForgeVersionDesc(a, b) {
+  if (typeof a === "string" && typeof b === "string") {
```
Comment (killpowa) [inline src/common/utils/index.js]: Codacy Issue found: [Replace `"string"·&&·typeof·b·===·"string"` with `'string'·&&·typeof·b·===·'string'`](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5243672)

**Comment by Mohron** [comment]
Sorry, haven't gotten my IDE synced up with the eslint/prettier rules and I did see the git hook changed the files.

**Review by blarfoon — APPROVED** [review]

**Pull request merged**